avocado/utils/cpu.py

```diff
@@ -420,3 +421,31 @@ def get_pid_cpus(pid):
         except IOError:
             continue
     return list(cpus)
+
+
+def _deprecated(newfunc, oldfuncname):
+    """
+    Print a warning to user and return the new function
+
+    :param newfunc: new function to be assigned
+    :param oldfunctionname: Old function name string
+    :rtype: `function`
+    """
+    fmt_str = "avocado.utils.cpu.{}() it is getting deprecat".format(oldfuncname)
+    fmt_str += "ed, Use avocado.utils.cpu.{}() instead".format(newfunc.__name__)
+    warnings.warn((fmt_str), DeprecationWarning, stacklevel=3)
+    if not hasattr(oldfuncname, '__call__'):
```

Comment:
Do you have a reason to check a string for `__call__` presence or is it a remnant of your debugs?

Also how about something like this:

```
diff --git a/avocado/utils/cpu.py b/avocado/utils/cpu.py
index 38bde5c7..6d02ff6f 100644
--- a/avocado/utils/cpu.py
+++ b/avocado/utils/cpu.py
@@ -431,12 +431,12 @@ def _deprecated(newfunc, oldfuncname):
     :param oldfunctionname: Old function name string
     :rtype: `function`
     """
-    fmt_str = "avocado.utils.cpu.{}() it is getting deprecat".format(oldfuncname)
-    fmt_str += "ed, Use avocado.utils.cpu.{}() instead".format(newfunc.__name__)
-    warnings.warn((fmt_str), DeprecationWarning, stacklevel=3)
-    if not hasattr(oldfuncname, '__call__'):
-        return newfunc
-
+    def wrap(*args, **kwargs):
+        fmt_str = "avocado.utils.cpu.{}() it is getting deprecat".format(oldfuncname)
+        fmt_str += "ed, Use avocado.utils.cpu.{}() instead".format(newfunc.__name__)
+        warnings.warn(fmt_str, DeprecationWarning, stacklevel=2)
+        return newfunc(*args, **kwargs)
+    return wrap
 
 total_cpus_count = _deprecated(total_count, "total_cpus_count")
 _get_cpu_info = _deprecated(_get_info, "_get_cpu_info")
```

to only issue the warnings of individual calls rather than warn on each `cpu` import?

Reply:
> Do you have a reason to check a string for `__call__` presence or is it a remnant of your debugs?
@ldoktor 
added to make sure to return once when there was no `__call__` attribute...

Thanks for your suggestion, it works well during call,

```
>>> from avocado.utils import cpu
>>> cpu.total_cpus_count()
__main__:1: DeprecationWarning: avocado.utils.cpu.total_cpus_count() it is getting deprecated, Use avocado.utils.cpu.total_count() instead
4
>>> cpu.total_cpus_count()
4
>>> cpu.total_cpus_count()
4
>>> 
```

Reply:
done, tnx